src/test/java/com/fittracker/fittracker/controller/ProductControllerValidationTest.java

```diff
@@ -0,0 +1,174 @@
+import static com.fittracker.fittracker.dataprovider.Request.productRequestWithKcal;
+import static com.fittracker.fittracker.dataprovider.Request.productRequestWithName;
+import static com.fittracker.fittracker.dataprovider.Request.productRequestWithProtein;
+import static org.junit.jupiter.params.provider.Arguments.of;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.springframework.context.annotation.FilterType.ASSIGNABLE_TYPE;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fittracker.fittracker.exception.ErrorResponseMapper;
+import com.fittracker.fittracker.request.ProductRequest;
+import com.fittracker.fittracker.security.JwtAuthenticationFilter;
+import com.fittracker.fittracker.service.ProductService;
+import java.net.URI;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.verification.VerificationMode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.web.servlet.MockMvc;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+
+@AutoConfigureMockMvc(addFilters = false)
+@Import(ErrorResponseMapper.class)
+@WebMvcTest(controllers = ProductController.class, excludeFilters = {
+    @ComponentScan.Filter(type = ASSIGNABLE_TYPE, value = JwtAuthenticationFilter.class)})
+public class ProductControllerValidationTest {
+
+    @MockBean
+    private ProductService productService;
+    @Autowired
+    private MockMvc mockMvc;
+
+    private static final String ENDPOINT = "/api/v1/product";
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    @BeforeEach
+    void beforeEach() {
+        mapper.setVisibility(FIELD, ANY);
+    }
+
+    @Nested
+    class Get {
```

Comment:
What happens if we sent a GET request to `/api/v1/product`?

Reply:
We get `HttpRequestMethodNotSupportedException` as indeed for `/api/v1/product` only mapping is done for POST method. Should I define @GetMapping for `/api/v1/product` and return there an error message?

Reply:
On second thought, I should add this case to tests and check if proper exception is thrown.